lib/OnyxUtils.ts

```diff
@@ -426,19 +432,17 @@ function getCachedCollection<TKey extends CollectionKeyBase>(collectionKey: TKey
 function keysChanged<TKey extends CollectionKeyBase>(
     collectionKey: TKey,
     partialCollection: OnyxCollection<KeyValueMapping[TKey]>,
-    previousPartialCollection: OnyxCollection<KeyValueMapping[TKey]> | undefined,
+    partialPreviousCollection: OnyxCollection<KeyValueMapping[TKey]>,
     notifyRegularSubscibers = true,
     notifyWithOnyxSubscibers = true,
 ): void {
-    const previousCollectionWithoutNestedNulls = previousPartialCollection === undefined ? {} : (utils.removeNestedNullValues(previousPartialCollection) as Record<string, unknown>);
-
     // We prepare the "cached collection" which is the entire collection + the new partial data that
     // was merged in via mergeCollection().
     const cachedCollection = getCachedCollection(collectionKey);
-    const cachedCollectionWithoutNestedNulls = utils.removeNestedNullValues(cachedCollection) as Record<string, unknown>;
+    const previousCollection = partialPreviousCollection ?? {};
 
     // If the previous collection equals the new collection then we do not need to notify any subscribers.
-    if (previousPartialCollection !== undefined && deepEqual(cachedCollectionWithoutNestedNulls, previousCollectionWithoutNestedNulls)) {
+    if (partialPreviousCollection !== undefined && deepEqual(cachedCollection, partialPreviousCollection)) {
```

Comment:
Do we need this check for `undefined`? The types suggested the `partialPreviousCollection` suggest it cannot be undefined so one of these feels incorrect